tool/src/dart/formatter.rs

```diff
@@ -18,16 +18,22 @@ use std::borrow::Cow;
 pub(super) struct DartFormatter<'tcx> {
     c: CFormatter<'tcx>,
     docs_url_generator: &'tcx DocsUrlGenerator,
+    strip_prefix: Option<String>,
 }
 
 const INVALID_METHOD_NAMES: &[&str] = &["new", "static"];
 const INVALID_FIELD_NAMES: &[&str] = &["new", "static"];
 
 impl<'tcx> DartFormatter<'tcx> {
-    pub fn new(tcx: &'tcx TypeContext, docs_url_generator: &'tcx DocsUrlGenerator) -> Self {
+    pub fn new(
+        tcx: &'tcx TypeContext,
+        docs_url_generator: &'tcx DocsUrlGenerator,
+        strip_prefix: Option<String>,
```

Comment:
is this so that we can strip the ICU4X prefix? I guess that works

The way I want to support this is to do it the other way; we _add_ prefixes in the C code and everyone else uses namespaces for this if they want to.

Reply:
But this does involve tweaking the existing backends; which is annoying. Ideally I'd like to abandon the AST backends soon so that we can start moving to the right architecture for stuff like this.

Reply:
> The way I want to support this is to do it the other way; we add prefixes in the C code and everyone else uses namespaces for this if they want to.

Yeah same. That does require adding annotations on the diplomat-bridge level, which is more complex at the moment.